pkg/plugins/runtime/gateway/plugin.go

```diff
@@ -34,6 +38,28 @@ func (p *plugin) BeforeBootstrap(context *core_plugins.MutablePluginContext, con
 	return nil
 }
 
+func (p *plugin) Apply(ctx context.Context, meshContext xds_context.MeshContext, proxy *core_xds.Proxy) error {
+	if proxy.Dataplane == nil || !proxy.Dataplane.Spec.IsBuiltinGateway() {
+		return nil
+	}
+	l, err := gatewayListenerInfoFromProxy(
+		ctx, &meshContext, proxy,
+	)
+	if err != nil {
+		return err
+	}
+	proxy.RuntimeExtensions[metadata.PluginName] = l
+	return nil
+}
+
+func ExtractGatewayListener(proxy *core_xds.Proxy) []GatewayListenerInfo {
+	ext := proxy.RuntimeExtensions[metadata.PluginName]
+	if ext == nil {
+		return nil
```

Comment:
shouldn't this function return empty list instead of nil?

Reply:
I'm always unsure with arrays in go as empty array and nil is roughly the same thing...

Reply:
Yeah it's functionnaly equivalent: https://github.com/golang/go/wiki/CodeReviewComments#declaring-empty-slices
I think nil is easier to read